nixos/services/mail/default.nix

```diff
@@ -200,7 +215,27 @@ in {
               inherit (role) mailHost webmailHost;
             });
           } config.fclib.network.fe.dualstack.addressesQuoted);
-        in listToAttrs (map cfgForDomain role.domains);
+        in listToAttrs ((map cfgForDomain role.domains) ++
+          [ (lib.nameValuePair fqdn
```

Comment:
This must point to the `role.mailHost`, not `fqdn` here. (it's a bit misleading in the code because `mailserver.fqdn = role.mailHost` but there's a `fqdn` in a let that points to the internal hostname like test33.fcio.net).

After fixing this, there will be a problem with nixos/services/mail/roundcube.nix, line 51 which redirects `role.mailHost` to the webmailer. This has to be removed. Also, we must assert that `role.webmailHost != role.mailHost` to avoid a conflict between imprint and webmailer...

Reply:
New plan: keep the redirect to webmailHost as platform default, but it can be overridden by setting imprintText or imprintUrl. Both options should be optional and not generate a warning. If mailHost == webmailHost, both options should produce an error.
We have to document this properly in the mailserver role docs.